src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2092,17 +2104,33 @@ public override void AfterTransactionBegin(ITransaction tx)
 			using (new SessionIdLoggingContext(SessionId))
 			{
 				CheckAndUpdateSessionStatus();
-				Interceptor.AfterTransactionBegin(tx);
+
+				if (!_transactionCoordinatorShared || Interceptor != ConnectionManager.Session.Interceptor)
```

Comment:
Why you do not call interceptors for "child" sessions? I think it is perfectly valid scenario.